test/astarte_vmq_plugin_connection_synchronizer_test.exs

```diff
@@ -0,0 +1,165 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.VMQ.Plugin.Connection.SynchronizerTest do
+  use ExUnit.Case
+
+  alias AMQP.{Channel, Connection, Queue}
+  alias Astarte.VMQ.Plugin.Config
+  alias Astarte.VMQ.Plugin.Connection.Synchronizer
+  alias Astarte.VMQ.Plugin.Connection.Synchronizer.Supervisor, as: SynchronizerSupervisor
+
+  @device_id :crypto.strong_rand_bytes(16) |> Base.url_encode64(padding: false)
+  @realm "test"
+  @device_base_path "#{@realm}/#{@device_id}"
+  @queue_name "#{Config.data_queue_prefix()}0"
+
+  setup_all do
+    amqp_opts = Config.amqp_options()
+    {:ok, conn} = Connection.open(amqp_opts)
+    {:ok, chan} = Channel.open(conn)
+    Queue.declare(chan, @queue_name)
+    {:ok, chan: chan}
+  end
+
+  setup %{chan: chan} do
+    test_pid = self()
+
+    {:ok, consumer_tag} =
+      Queue.subscribe(chan, @queue_name, fn payload, meta ->
+        send(test_pid, {:amqp_msg, payload, meta})
+      end)
+
+    on_exit(fn ->
+      Queue.unsubscribe(chan, consumer_tag)
+    end)
+
+    :ok
+  end
+
+  test "Standard connection flow" do
+    connection_timestamp = now_us_x10_timestamp()
+
+    # no reconciler
+    assert Registry.lookup(AstarteVMQPluginConnectionSynchronizer.Registry, @device_base_path) ==
+             []
+
+    # start the reconciler
+    reconciler_pid = get_connection_reconciler_process!(@device_base_path)
```

Comment:
Why are you manually starting the reconciler? Shouldn't `handle_connection` and/or `handle_disconnection` take care of starting it in tests?

Reply:
`handle_[dis]connection/3` receives  as a parameter the PID of a process which handles events, it does not create such a process.  So tests must create it beforehand